web/src/main/ui/app/components/flows-new/edit-flow/mapping/mapping.component.ts

```diff
@@ -148,19 +148,38 @@ export class MappingComponent implements OnInit {
         this.sampleDocURI = map.sourceURI;
         this.editURIVal = this.sampleDocURI;
       }
-      this.loadSampleDoc()
+      this.loadSampleDoc();
       if (map && map.properties) {
+        this.isNestedMap();
+        console.log("isNested: " + this.connsNested);
         self.conns = {};
-        _.forEach(map.properties, function(srcObj, entityPropName) {
-          self.conns[entityPropName] = srcObj.sourcedFrom;
-        });
+        if(!self.connsNested ) {
+          _.forEach(map.properties, function(srcObj, entityPropName) {
+            self.conns[entityPropName] = srcObj.sourcedFrom;
+          });
+        }
+        else {
+          self.conns = map.properties;
+        }
         self.connsOrig = _.clone(self.conns);
       }
     },
     () => {},
     () => {});
   }
 
+  isNestedMap() {
+    let props = Object.keys(this.mapping.properties)
+    for(let i = 0; i < props.length ; i++){
```

Comment:
Just a note, I think this can be simplified by using an array function, such as "find" or "findIndex":

    return this.mapping.properties.findIndex(prop => prop.hasOwnProperty('targetEntityType') > -1;

Reply:
@rjrudin ,
'findIndex' is available for Arrays only. When I use it, I get  this.mapping.properties.findIndex is not a function

Reply:
Oh sorry - I mistyped that, it would be Object.keys(this.mapping.properties).findIndex(key => this.mapping.properties[key].hasOwnProperty("targetEntityType"). Or something like that - just pointing out that any time you write a for loop with an index counter, there's likely an array function that gets it done in one line.

Reply:
Fixed in latest commit